### DISCUSSION TITLE
Improving documentation

### DISCUSSION THREAD

**Discussion by arnaudvergnet** [organicmaps/organicmaps#2064] [Q&A]
I looked through the [`/docs`](https://github.com/organicmaps/organicmaps/tree/master/docs) folder but I find the documentation there is a bit lacking.

The repo is quite large and there are no explanations (at least I did not find any) as to what each folder contains. There are also no explanations about the architecture for the c++ backend or the apps (iOS, Android and Desktop). Same for editing styles or the generator I could not find any documentation.

If there is some documentation I missed, please link me to it but in any case I think it is a bit difficult for new contributors to find this information. Improving documentation would make community contributions easier and greatly help the project. 

I don't know if I should create an issue for this so it will be a discussion for now.

**Comment by biodranik** [comment]
@pastk can help here.

**Comment by pastk** [comment]
Yeap there is a lot to improve indeed, some of the work had been done here already https://github.com/organicmaps/organicmaps/discussions/1538

Directories structure is briefly covered at https://github.com/organicmaps/organicmaps/blob/master/docs/STRUCTURE.md though needs much more detail.

A guide to generate maps is at https://github.com/organicmaps/organicmaps/blob/master/tools/python/maps_generator/README.md (never followed it so IDK if its up-to-date).

I plan to add/update docs re styling and translations soonish.

> I think it is a bit difficult for new contributors to find this information. Improving documentation would make community contributions easier and greatly help the project.

100% agreed!

**Reply by arnaudvergnet** [reply]
Would love to help but I am one of those new contributors needing the documentation

**Reply by pastk** [reply]
Here is another batch: #2196
You can help by reviewing if you want!

**Comment by arnaudvergnet** [comment]
> Directories structure is briefly covered at https://github.com/organicmaps/organicmaps/blob/master/docs/STRUCTURE.md though needs much more detail.

is it possible to simplify the directory structure? At least at the root level. Grouping folders of the same type would make understanding the structure easier. For example, grouping the folders `android`, `iphone` and `qt` into a folder named `ui` or `apps` makes sense. Same for `routing`, `search` and others into a folder `cpp_backend`.

This way it would be easier to understand even with few documentation.

**Reply by pastk** [reply]
I would love it too :)
The biggest downside is probably it'll mess the git history so it'll be hard to search.
@biodranik ?

**Reply by biodranik** [reply]
No, we won't break the structure, sorry. It will break all scripts too :)
